qt/applications/workbench/workbench/app/main.py

```diff
@@ -0,0 +1,221 @@
+from workbench.config import APPNAME, ORG_DOMAIN, ORGANIZATION
+from workbench.plugins.exception_handler import exception_logger
+from workbench.widgets.about.presenter import AboutPresenter
+
+from qtpy.QtGui import QIcon
+from qtpy.QtWidgets import QApplication
+from qtpy.QtCore import QCoreApplication, Qt, qVersion
+
+SYSCHECK_INTERVAL = 50
+ORIGINAL_SYS_EXIT = sys.exit
+ORIGINAL_STDOUT = sys.stdout
+ORIGINAL_STDERR = sys.stderr
+STACKTRACE_FILE = 'workbench_stacktrace.txt'
+
+
+def qapplication():
+    """Either return a reference to an existing application instance
+    or create a new one
+    :return: A reference to the QApplication object
+    """
+    app = QApplication.instance()
+    if app is None:
+        QCoreApplication.setAttribute(Qt.AA_ShareOpenGLContexts)
+        argv = sys.argv[:]
+        argv[0] = APPNAME  # replace application name
+        # Workaround a segfault with the IPython console when using Python 3.5 + PyQt 5
+        # Without this using this fix the above combination causes a segfault when the IPython
+        # console is started
+        # The workaround mentioned in https://groups.google.com/forum/#!topic/leo-editor/ghiIN7irzY0
+        # is to ensure readline is imported before the QApplication object is created
+        if sys.version_info[0] == 3 and sys.version_info[1] == 5:
+            importlib.import_module("readline")
+        app = QApplication(argv)
+        app.setOrganizationName(ORGANIZATION)
+        app.setOrganizationDomain(ORG_DOMAIN)
+        app.setApplicationName(APPNAME)
+        app.setApplicationVersion(mantid_version_str())
+        # Spin up the usage service and set the name for the usage reporting
+        # The report is sent when the FrameworkManager kicks up
+        UsageService.setApplicationName(APPNAME)
+
+        if is_required_version(required_version='5.10.0', version=qVersion()):
+            app.setAttribute(Qt.AA_DisableWindowContextHelpButton)
+
+    return app
+
+
+def initialize():
+    """Perform an initialization of the application instance.
+
+        - Patches sys.exit so that it does nothing.
+        - Uses WindowsSelectorEventLoop required by Tornado
+
+    :return: A reference to the existing application instance
+    """
+    if sys.version_info.major >= 3 and sys.platform == 'win32':
+        # Tornado requires WindowsSelectorEventLoop
+        # https://www.tornadoweb.org/en/stable/#installation
+        import asyncio
+        asyncio.set_event_loop_policy(asyncio.WindowsSelectorEventLoopPolicy())
+
+    app = qapplication()
+
+    # Monkey patching sys.exit so users can't kill
+    # the application this way
+    def fake_sys_exit(arg=[]):
+        pass
+
+    sys.exit = fake_sys_exit
+
+    return app
+
+
+def start_workbench(app, command_line_options):
+    """Given an application instance create the MainWindow,
+    show it and start the main event loop
+    """
+    from workbench.app.mainwindow import MainWindow
```

Comment:
Could we maybe add a comment that this import needs to be kept local to avoid importing the matplotlib backend too early? A future developer may not understand why all of the other imports are at the top of the module.